### PR TITLE
Update isotopologues when changing atom types in the GUI

### LINKED ISSUE #956: Changing AtomTypes doesn't update isotopologues
**Describe the bug**
Changing atom types in a species does not update the isotopologues.

### PR DESCRIPTION
**Pull request by rprospero**
This only updates the isotopologues when the changes are in the GUI, but I believe that is the only place where this can occur.

Closes #956

### CODE REVIEW

**Review by trisyoungs — APPROVED** [review]
Looks good - one small suggestion for improving the intent/clarity of the model in the comments.

I believe you're right that this is the only place where changing atom types should trigger update of the isotopologues. Updates should also happen when editing / deleting atom types from the `Forcefield` page, but I'm not entirely sure that's currently the case.

File: src/gui/models/speciesAtomModel.cpp [hunk 1]
```diff
@@ -1,7 +1,10 @@
 #include "gui/models/speciesAtomModel.h"
 #include "classes/atomtype.h"
 
-SpeciesAtomModel::SpeciesAtomModel(std::vector<SpeciesAtom> &atoms, Dissolve &dissolve) : atoms_(atoms), dissolve_(dissolve) {}
+SpeciesAtomModel::SpeciesAtomModel(std::vector<SpeciesAtom> &atoms, Species &species, Dissolve &dissolve)
```
[the remark this answers is not in the document]
Comment (rprospero) [reply src/gui/models/speciesAtomModel.cpp]: I'd had the same thought, but hadn't followed through on it.  The code looks cleaner after the change.

**Comment by rprospero** [comment]
From a cursory glance, it seems that the Forcefield tab can modify AtomTypes and add them to the overall Dissolve simulation, but it cannot change the species, so I don't believe that we need to update the isotopologues.  However, if I'm wrong and this does require an update, I can connect that to the dataChanged event pretty simply.

**Comment by trisyoungs** [comment]
It was the deletion of atom types that I was wondering about, as that propagates potential changes in Species and their Isotopologues. However, that button is `NOT IMPLEMENTED YET.` so there's nothing to worry about for now!

**Pull request merged**